zerver/migrations/0417_backfill_message_realm.py

```diff
@@ -0,0 +1,75 @@
+from django.db import migrations, models, transaction
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def backfill_message_realm(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    RECIPIENT_HUDDLE = 3
+
+    Message = apps.get_model("zerver", "Message")
+    Realm = apps.get_model("zerver", "Realm")
+    Stream = apps.get_model("zerver", "Stream")
+    Subscription = apps.get_model("zerver", "Subscription")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    BATCH_SIZE = 10000
+
+    if not Message.objects.exists():
+        # Nothing to do if there are no messages yet.
+        return
+
+    def set_realm_batched(query: models.query.QuerySet[Message], realm_id: int) -> int:
+        count = 0
+        while True:
+            with transaction.atomic():
+                message_ids = set(
+                    query.filter(realm=None)
+                    .select_for_update()
+                    .values_list("id", flat=True)[:BATCH_SIZE]
+                )
+                if not message_ids:
+                    break
+                count += len(message_ids)
+                Message.objects.filter(id__in=message_ids).update(realm_id=realm_id)
```

Comment:
Do you think it's worth adding a print after processing a batch that was large enough that `count >= BATCH_SIZE`? That would provide a useful progress indicator in a server containing a large single realm, while not producing a bunch of spam for small realms.

Reply:
Makes sense, added a print